content/articles/spring-webflux/index.md

```diff
@@ -0,0 +1,380 @@
+    
+```java
+public interface StudentRepository extends ReactiveMongoRepository<Student, Integer> {
+    @Query("{ 'name': ?0 }")
+    Flux<Student> findByName(final String name); // Flux returns zero or n elements
+}
+
+```
+
+### Service layer
+1. In the root project package, create a new package named `service`.
+2. Create a file named `StudentService.java` and add the code snippet below. 
+```java
+public interface StudentService {
+    void createStudent(Student student); // Returns null after creating a student
+
+    Mono<Student> findById(int id); // Returns 0 or a single student
+
+    Flux<Student> findByName(String name); // Returns a list of students whose names match the searched name
+
+    Flux<Student> findAll(); // Returns all students
+
+    Mono<Student> update(Student student); // Updates and returns the updated student
+
+    Mono<Void> delete(int id); // Delete the student
+
+}
+```
+
+3. In the `service` package, create a new file named `StudentServiceImpl.java` and add the code snippet below.
+```java
+@Service
+@AllArgsConstructor
+public class StudentServiceImpl implements StudentService {
+    private final StudentRepository repository;
+    // Saves the student into the database
+    @Override
+    public void createStudent(Student student) {
+
+        repository.insert(student).subscribe();
+    }
+    // Finds a single student by id
+    @Override
+    public Mono<Student> findById(int id) {
+        return repository.findById(id);
+    }
+    // Finds a list of students whose names match the searched name
+    @Override
+    public Flux<Student> findByName(String name) {
+        return repository.findByName(name);
+    }
+    // Returns a list of all students from the database
+    @Override
+    public Flux<Student> findAll() {
+        return repository.findAll();
+    }
+    // Saves a student into the database
+    @Override
+    public Mono<Student> update(Student student) {
+        return repository.save(student);
+    }
+    // Deletes a student from the database
+    @Override
+    public Mono<Void> delete(int id) {
+        return repository.deleteById(id);
+    }
+}
+
+```
+
+### Controller layer
+1. In the root project package, create a new package named `controllers`.
+2. Create a new file named `StudentController.java` in the `controllers` package created above.
+```java
+@RestController // Marks this class as a REST controller
+@RequestMapping("/api/students") // Sets the base URL for students API
+@AllArgsConstructor // Lombok annotation to generates a constructor for the class
+public class StudentController {
+    private final StudentService service;
+    // Handles the student creation POST request.
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public void createStudent(Student student) {
+        service.createStudent(student);
+    }
+    // Handles get student by id endpoint
+    @GetMapping("/{id}")
```

Comment:
You forgot to change this path to remove the ambiguous request mapping error. I will be fixing this in the code in your article but please fix it in your `repl.it` repo. It is not fixed in the `repl.it` repository either.

I'm going to change the path to get the student by id to `/api/students/id/{id}` and that for the name to be `/api/students/name/{name}`, as you state near the end of the article.

Reply:
My bad, I realized my forked copy of your replit repo was outdated, you did fix it in the replit repo. Sorry!